### PR TITLE
Adds editorconfig for current code style

### PR DESCRIPTION
**Pull request by ericdecanini**
Draft for testing with the linter on the CI

## Type of change

- [ ] Feature
- [ ] Bugfix
- [x] Technical
- [ ] Other :

## Content

Adds an `.editorconfig` file reflecting the current code style

## Motivation and context

Closes https://github.com/vector-im/element-android/issues/5729

In order to make incremental changes to our current code style, it's best for us to have an initial editorconfig that reflects our current code style that we then make changes to later.

An issue encountered when doing this is there are areas of the project that don't follow even the current code style for reasons such as indentation, wrapping, etc.

Instead of changing these files which would result in a massive diff, I opted for disabling these rules temporarily from the linter for the duration we undergo this code style change.

The editorconfig file found here is one extracted from Android Studio preferences reflecting the current code style. It includes settings for other languages too other than Kotlin. At first I thought we didn't need this and removed them but this resulted in a bug where XML files were unintentionally formatted differently so I decided to keep them all

## Screenshots / GIFs

N/A

## Tests

The main test here in that the linter passes. See that we have no changes to our autoformatter for different file types (kotlin, xml)

I also tested the app doesn't crash on boot

## Tested devices

- [ ] Physical
- [x] Emulator
- OS version(s): Android 12

## Checklist

- [ ] Changes has been tested on an Android device or Android emulator with API 21
- [ ] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#accessibility
- [x] Pull request is based on the develop branch
- [x] Pull request includes a new file under ./changelog.d. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#changelog
- [ ] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [x] You've made a self review of your PR
- [ ] If you have modified the screen flow, or added new screens to the application, you have updated the test [UiAllScreensSanityTest.allScreensTest()](https://github.com/vector-im/element-android/blob/main/vector/src/androidTest/java/im/vector/app/ui/UiAllScreensSanityTest.kt#L73)

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -1,14 +1,901 @@
...
+ij_java_wrap_first_method_in_call_chain = false
+ij_java_wrap_long_lines = false
+
+[*.properties]
+ij_properties_align_group_field_declarations = false
+ij_properties_keep_blank_lines = false
+ij_properties_key_value_delimiter = equals
+ij_properties_spaces_around_key_value_delimiter = false
+
+[.editorconfig]
```
Comment (bmarty) [inline .editorconfig:256]: inception!

File: .editorconfig [hunk 2]
```diff
@@ -1,14 +1,901 @@
...
-# true (recommended) / false
-insert_final_newline=true
-# possible values: number (e.g. 120) (package name, imports & comments are ignored), "off"
-# it's automatically set to 100 on `ktlint --android ...` (per Android Kotlin Style Guide)
-max_line_length=off
-
-# From https://github.com/pinterest/ktlint#custom-ktlint-specific-editorconfig-properties
-# default IntelliJ IDEA style, same as alphabetical, but with "java", "javax", "kotlin" and alias imports in the end of the imports list
-ij_kotlin_imports_layout=*,java.**,javax.**,kotlin.**,^
+[*]
```
Comment (bmarty) [inline .editorconfig:1]: what was the command to generate this file OOI?

Comment (ericdecanini) [reply .editorconfig:1]: image

;)

Comment (bmarty) [reply .editorconfig:1]: TIL, thanks!

**Review by fedrunov — APPROVED** [review]
LGTM

**Pull request merged**